instrumentation/github.com/aws/aws-sdk-go/service/otels3/mocks/s3.go

```diff
@@ -12,8 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// package mocks provides a mock interface to the S3 client
 package mocks
 
-type S3Client struct {
-	S3API
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+)
+
+type MockS3Client struct {
+	s3iface.S3API
+}
+
+// PutObjectWithContext implements the mocked S3 client successfully returning an empty PutObjectOutput type
+func (s *MockS3Client) PutObjectWithContext(ctx aws.Context, input *s3.PutObjectInput, opts ...request.Option) (*s3.PutObjectOutput, error) {
```

Comment:
It's mock, we don't really need to write a comment for mock I think. But it's nice to have 👍